### PR TITLE
Add Functionality

### PR DESCRIPTION
**Pull request by Snayak7643**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,69 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import Card from "./Components/Card";
+import Loading from "./Components/Loading";
+import Refresh from "./Refresh";
+
+type Tour = {
```
Comment (Ishaan-flyfin) [inline src/App.tsx]: nice!

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,69 @@
...
 function App() {
-  return <h1>Tours</h1>;
+  const url = "https://course-api.com/react-tours-project";
+
+  /* States */
+  const [tours, setTours] = useState<Tour[]>([]);
+  const [isLoading, setLoading] = useState<boolean>();
+
+  /* Fetching Function */
+  const fetching = async () => {
```
Comment (Ishaan-flyfin) [inline src/App.tsx]: we can give a better name
something which tells what we are fetching here

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,69 @@
...
+    setLoading(false);
+  };
+
+  /* Function For Not Interested(Removing Tour) */
+  const deleteTour = (id: string) => {
+    const newTours = tours.filter((tour) => {
+      return id !== tour.id;
+    });
+    setTours(newTours);
+  };
```
Comment (Ishaan-flyfin) [inline src/App.tsx:30]: nice

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,69 @@
...
+
+  /*  */
+  useEffect(() => {
+    fetching();
+  }, []);
```
Comment (Ishaan-flyfin) [inline src/App.tsx]: here we can think of usecallback usage
lets discuss this in call

File: src/App.tsx [hunk 5]
```diff
@@ -1,7 +1,69 @@
...
+  const url = "https://course-api.com/react-tours-project";
+
+  /* States */
+  const [tours, setTours] = useState<Tour[]>([]);
+  const [isLoading, setLoading] = useState<boolean>();
+
+  /* Fetching Function */
+  const fetching = async () => {
+    setLoading(true);
+    const response = await fetch(url);
```
Comment (Ishaan-flyfin) [inline src/App.tsx:18]: we can add api error handling also

File: src/Components/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+type PropType = {
+  tour: {
+    id: string;
+    image: string;
+    info: string;
+    name: string;
+    price: string;
+  };
```
Comment (Ishaan-flyfin) [inline src/Components/Card.tsx]: we can create a file `types.ts`
in which we can export reusable types like tour which is being used in app.tsx and card.tsx

File: src/Components/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  deleteTour: (id: string) => void;
+};
+
+const Card: React.FunctionComponent<PropType> = ({ tour, deleteTour }) => {
```
Comment (Ishaan-flyfin) [inline src/Components/Card.tsx:9]: instead  of tour, we can pass specific props which makes this component more reusable

**Review by Ishaan-flyfin — CHANGES_REQUESTED** [review]
nice progress!
added some minor comments

File: src/Components/Card.tsx (hunk 6 above)
Comment (Snayak7643) [reply src/Components/Card.tsx]: Done

File: src/App.tsx (hunk 5 above)
Comment (Snayak7643) [reply src/App.tsx:18]: Done

File: src/App.tsx (hunk 2 above)
Comment (Snayak7643) [reply src/App.tsx]: Renamed to fetchingTours

**Review by Ishaan-flyfin — APPROVED** [review]

**Pull request merged**